### PR TITLE
[release-4.12] OCPBUGS-13790: Address potential unitialized variables

### PR DESCRIPTION
**Pull request by sebsoto**

### CODE REVIEW

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

File: pkg/nodeconfig/init.go [hunk 1]
```diff
@@ -39,7 +39,6 @@ func init() {
 	kubeAPIServerEndpoint, err := discoverKubeAPIServerEndpoint()
 	if err != nil {
 		log.Error(err, "unable to find kube api server endpoint")
-		return
```
Comment (aravindhp) [inline pkg/nodeconfig/init.go:42]: Isn't it safer to return?

File: pkg/nodeconfig/nodeconfig.go [hunk 2]
```diff
@@ -129,6 +129,14 @@ func NewNodeConfig(c client.Client, clientset *kubernetes.Clientset, clusterServ
 		return nil, fmt.Errorf("error instantiating Windows instance from VM: %w", err)
 	}
 
+	// initialize the cache now if it failed to initialize
```
Comment (aravindhp) [inline pkg/nodeconfig/nodeconfig.go:132]: Don't we need to do the same for `nodeConfigCache.credentials`?

**Review by aravindhp — COMMENTED** [review]
Thanks for the quick turnaround on this, @sebsoto

**Review by aravindhp — COMMENTED** [review]
Sorry I missed the 2nd commit

File: pkg/nodeconfig/nodeconfig.go (hunk 2 above)
Comment (aravindhp) [reply pkg/nodeconfig/nodeconfig.go:132]: Sorry I missed the 2nd commit

File: pkg/nodeconfig/init.go (hunk 1 above)
Comment (sebsoto) [reply pkg/nodeconfig/init.go]: thats probably for the best

**Pull request merged**